src/compiler/checker.ts

```diff
@@ -7173,6 +7173,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
             else {
+                if(propertySymbol.flags & SymbolFlags.Accessor && propertySymbol.declarations && propertySymbol.declarations.length === 2) {
+                    const getter = isGetAccessor(propertySymbol.declarations[0]) ? propertySymbol.declarations[0] : isGetAccessor(propertySymbol.declarations[1]) ? propertySymbol.declarations[1] : undefined;
+                    const setter = isSetAccessor(propertySymbol.declarations[0]) ? propertySymbol.declarations[0] : isSetAccessor(propertySymbol.declarations[1]) ? propertySymbol.declarations[1] : undefined;
+                    if(getter && setter) {
+                        const getterReturnType = getter.type && getTypeFromTypeNode(getter.type);
+                        let setterParameterType: Type | undefined;
+                        if(setter.parameters.length === 1) {
+                            setterParameterType = setter.parameters[0].type && getTypeFromTypeNode(setter.parameters[0].type);
+                        }
+                        else if (setter.parameters.length === 2 && isThisIdentifier(setter.parameters[0].name)) {
+                            setterParameterType = setter.parameters[1].type && getTypeFromTypeNode(setter.parameters[1].type);
+                        }
+                        if(setter.parameters.length > 2 || (setterParameterType && getterReturnType && !isTypeIdenticalTo(setterParameterType, getterReturnType))) {
+                            typeElements.push(factory.createGetAccessorDeclaration(/*modifiers*/ undefined, getter.name, getter.parameters, /*type*/ getter.type, /*body*/ undefined));

```

Comment:
The `type` node here and on the parameter in the set accessor needs to have its existence guaranteed (we should add it if it's missing) and traversed to check for unserializable types (if, eg, they refer to nodes not present in the enclosing declaration). `serializeTypeForDeclaration` should do all of this for you, including reusing the input type nodes if possible.

Reply:
Hi @weswigham , thanks for your code review because I am pretty new to TypeScript source code, there are some dull questions I want to ask 
- what is the difference between `Type` and `TypeNode` data structure in TypeScript, My personal understanding is Type is not AST, but TypeNode is.
- what is the concept of `serialize` in `serializeTypeForDeclaration`, is just transforming `Type` to `TypeNode` ?

Sorry about asking those dull questions, thanks for any reply

Reply:
Hi, @weswigham , I think that `serializeTypeForDeclaration` would still narrow setter type into a number even when the `type` property of setter parameter is a union (`number | string`), I use it as below :
```ts
const setterParameterType = getTypeFromTypeNode(serializeTypeForDeclaration(context, propertyType, propertySymbol, saveEnclosingDeclaration))
```
If there is any misunderstanding about this function, please tell me , and thanks your code review again